Adding google scholar field and 2 other without url verification

Suryansh5545 opened the pull request:
#### Changes proposed in this pull request: Added Google scholar and other field in profile page

<!-- Changes: Add here what changes were made in this issue and if possible provide links. -->
-added 3 input field for google scholar , linkedin , github.


<!-- Screenshots for the change: Add here the screenshot of the fix. -->
-
![evalai-ngx profile](https://user-images.githubusercontent.com/34577232/70541628-60682c80-1b8d-11ea-85ff-d1c615ad3da8.png)

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html
	modified:   src/app/components/profile/profile.component.scss
	modified:   src/app/components/profile/profile.component.ts

yashdusing commented:
@Suryansh5545 the main EvalAI repo segregates the fields to two columns. Can you try doing the same for consistency purposes
![download](https://user-images.githubusercontent.com/19976688/70542558-ec2e8880-1b8e-11ea-8d80-7b8c2cbb4bf1.png)

Suryansh5545 commented:
I was able to achieve this . is this ok?

![evalai-ngx-profile](https://user-images.githubusercontent.com/34577232/70546627-a32e0280-1b95-11ea-975f-a01cf2549f77.png)

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html
	modified:   src/app/components/profile/profile.component.scss

pushkalkatara commented on src/app/components/profile/profile.component.html:
```diff
@@ -14,32 +13,57 @@
 			<div class="item">
 				<span class="text-light-black fw-semibold fs-12">
-					Affiliated to<br>
+					Lindedin Url<br>
```
please change the text to "LinkedIn URL".

Suryansh5545 committed:
Update profile.component.html

Suryansh5545 commented:
I have changed it sir

pushkalkatara commented on src/app/components/profile/profile.component.html:
```diff
@@ -48,18 +72,18 @@
 
 				<div class="modal-container" [class.hidden]="!isTokenModalVisible">
-					<div class="modal-card">
+					<div class="modal-card" >
```
please remove the extra space.

Suryansh5545 committed:
Update profile.component.html

Suryansh5545 committed:
Update profile.component.scss

pushkalkatara commented on src/app/components/profile/profile.component.ts:
```diff
@@ -171,6 +171,30 @@ export class ProfileComponent implements OnInit {
           value: this.user['affiliation']
+        },
+        {
+          isRequired: true,
```
Can we make the field optional? Maybe the user doesn't have a Google Scholar URL.

pushkalkatara commented:
@Suryansh5545 As @yashdusing suggested, we would need the UI to be consistent as it was previously. Try to keep the same 3 column format and add the new fields.

Suryansh5545 committed:
Update profile.component.ts

Suryansh5545 commented:
So i have to make sure that these field are in a card style container just like the current version of evalai?

pushkalkatara commented:
Yes, you got that right.

Suryansh5545 commented:
i put them in card like style but it is with static margins . i tried make page dynamic but i wasn't successful .
![evaliai-field change ](https://user-images.githubusercontent.com/34577232/70611735-ad024500-1c2b-11ea-8982-691a2d73099e.png)

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html
	modified:   src/app/components/profile/profile.component.scss

Suryansh5545 committed:
Merge branch 'master' of https://github.com/Suryansh5545/EvalAI-ngx

pushkalkatara commented:
Hi @Suryansh5545 You can remake the whole component again. There are several examples available [online](https://github.com/creativetimofficial/material-dashboard-angular2/blob/master/src/app/user-profile/user-profile.component.html).

pushkalkatara commented on src/app/components/profile/profile.component.html:
```diff
@@ -1,10 +1,11 @@
 <app-header-static></app-header-static>
-
 <div class="mid-container">
```
hint: we have to format the holder class for dynamic alignment to work.

Suryansh5545 commented:
I have rebuild component it is more dynamic then before look almost same but i still cant get the card to get full screen when the app-side-bar collapse because of short screen. please help me
![evalai ngx profile2](https://user-images.githubusercontent.com/34577232/70712282-73077080-1d09-11ea-8664-47aaafb9d752.png).

When screen app-side-bar collapse then it look like this 
![evalai ngx profile3](https://user-images.githubusercontent.com/34577232/70712351-99c5a700-1d09-11ea-8a9e-10cafda6a7a1.png)

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html
	modified:   src/app/components/profile/profile.component.scss

Sanji515 commented:
Hey @Suryansh5545 You need to set `width` to `100%` of right container for smaller devices. Please let me know if you need some more hints :smiley:

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.scss

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html

Suryansh5545 commented:
Successfully made the document work and look like evalai profile page also i have removed and check for empty line and wrong indent.  if require more changes please let me know

Sanji515 reviewed (commented):
@Suryansh5545 Can you please try to make the UI similar to as in the current version (https://evalai.cloudcv.org/web/profile) like giving some margin to the bottom, changing the back-ground color and placement of `change password` and `update profile` button and also some other minor adjustment.

Sanji515 reviewed (commented):
Also I think these two columns are not aligned properly. Please fix this.

![Screenshot from 2019-12-12 23-59-20](https://user-images.githubusercontent.com/32524438/70738969-d01c1a00-1d3b-11ea-8e83-8cc4159314e6.png)

Suryansh5545 commented:
Sir i have made it look almost similar to current evalai profile page with everything in place. an also have checked for empty line and indent
![evalai ngx profile4](https://user-images.githubusercontent.com/34577232/70751315-1bdbbd00-1d56-11ea-9c41-fcb6931ae94e.png)

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html
	modified:   src/app/components/profile/profile.component.scss

Sanji515 commented on src/app/components/profile/profile.component.html:
```diff
@@ -1,101 +1,117 @@
-<app-footer></app-footer>
+	<app-footer class="container-fluid" ></app-footer>
+  </div>
+</div>
```
Please add a new line here.

Suryansh5545 replied:
sir like a div tag containing the app footer

Sanji515 replied:
no just a blank line

Suryansh5545 replied:
a blank space above app-footer

Sanji515 replied:
@Suryansh5545 No, I mean to add an extra new blank line which should be line number 151

Suryansh5545 replied:
i have added a line no 151 at the end

Sanji515 replied:
yes, a blank line.

Sanji515 commented on src/app/components/profile/profile.component.scss:
```diff
@@ -1,20 +1,18 @@
+#card_container {
+	
+.card {
+		background: $gray-lighter;
```
Can you please indent this properly?

Sanji515 commented on src/app/components/profile/profile.component.scss:
```diff
@@ -78,19 +73,138 @@
+    left: -200px;
+    text-align: center;
+    color: #fff;
+    /*transition:all 0.2s ease-in-out;*/
```
Please remove this and above comment.

Sanji515 commented on src/app/components/profile/profile.component.scss:
```diff
@@ -78,19 +73,138 @@
+	  margin-bottom: 20px;
+	  margin-top: 25px;
+ }
+.container-fluid{
```
Please give a space ` ` after class name here and at other places.

Sanji515 replied:
@Suryansh5545 Also I mean to give a space before `{` just after class name in this case after `container-fluid`. It should look like this `.container-fluid {`

Suryansh5545 replied:
done sir

Sanji515 commented on src/app/components/profile/profile.component.scss:
```diff
@@ -78,19 +73,138 @@
+	  width: 100%; } }
+#profile_layout{
+	margin-top: 20px;
+}
```
Please indent it properly.

Sanji515 commented on src/app/components/profile/profile.component.scss:
```diff
@@ -78,19 +73,138 @@
+
+
+
+
```
Please remove these extra blank lines.

Sanji515 commented on src/app/components/profile/profile.component.html:
```diff
@@ -1,101 +1,117 @@
+							<br>
+							<br>
+							<div class="item">
+								<button class="btn btn-waves-effect waves-dark grad-btn grad-btn-dark fs-16 rm-pad-top" (click)="isTokenModalVisible= true;"> Get your Auth Token</button>
```
Please format and make it max-line length to 80.

Sanji515 replied:
Here, I mean to make the line-length to <=80. For line number 28 it is going to 187. Please let me know if you didn't get this.

Sanji515 commented on src/app/components/profile/profile.component.html:
```diff
@@ -1,101 +1,117 @@
 							<li>
-								<span class="btn btn-waves-effect waves-dark grad-btn grad-btn-dark
-									fs-14 rm-pad-top" (click)="tokenModalButtonToggle()">{{tokenModalButton}}</span>
+								<button (click)="updateUserDetails()" class="btn btn-waves-effect waves-dark grad-btn grad-btn-dark rm-pad-top fs-16">Update Profile</button>
```
Same here, Max-line length should be 80.

Sanji515 reviewed (commented):
I unable to see the `Submit` and `cancel` button on `Update profile` modal. Please make it scroll-able.

Sanji515 reviewed (commented):
After clicking `Get Auth Token` nothing happens but the button color changes. Please fix this.

Sanji515 commented:
@Suryansh5545 `Change password` functionality also not working, Please look into it.

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html
	modified:   src/app/components/profile/profile.component.scss
	modified:   src/styles/base.scss

Suryansh5545 commented:
i have made the request changes

Sanji515 commented:
Great! @Suryansh5545 Can you please push the changes?

Suryansh5545 commented:
i have pushed them

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html

Sanji515 commented on src/app/components/profile/profile.component.scss:
```diff
@@ -1,96 +1,224 @@
+}
+#sty_button{
+	max-lines: 80;
+}
```
Please add a new blank line here as well.

Suryansh5545 replied:
done sir

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.scss

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.scss

Sanji515 commented on src/app/components/profile/profile.component.scss:
```diff
@@ -1,96 +1,207 @@
+	color:$gray-dark;
+	margin:30px;
+	.dark {
+		color:$gray-darker;
```
@Suryansh5545 It is not indented properly, can you please look into it?

Suryansh5545 replied:
i indented it properly

Sanji515 replied:
if you have made changes then please push I'll look into it.

Sanji515 commented on src/app/components/profile/profile.component.scss:
```diff
@@ -1,96 +1,207 @@
+}
+}
+.copy {
+	background: none !important;
```
It will look more better if you try to give a blank line between two CSS :smiley:

Suryansh5545 replied:
like after a css tag end?

Sanji515 replied:
yes, like here just after the `.modal-container` css tag end.

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.scss

Sanji515 commented:
@Suryansh5545 On the `Get Auth token` modal, `Copy token` button and `token input` should align on the same line, please take a look at the current version at (https://evalai.cloudcv.org/web/profile).

Sanji515 commented:
@Suryansh5545 You should make the `Update-profile` modal scroll-able.

Also on clicking `Get auth token` or `update profile` button, it's color changes. Please check.

Sanji515 commented on src/app/components/profile/profile.component.scss:
```diff
@@ -1,82 +1,88 @@
+	background: $gray-lighter;
+	padding-bottom:30px;
+	width:50%;
+	margin:0 auto;
```
I think you are missing here. Please give one tab space to these(6 to 10 line number) lines.

Sanji515 commented on src/app/components/profile/profile.component.scss:
```diff
@@ -1,82 +1,88 @@
+.modal-container {
+	&.hidden {
+	display:none;
+}
```
Indent this line (One tab space please.)

Sanji515 commented on src/app/components/profile/profile.component.scss:
```diff
@@ -1,82 +1,88 @@
+		color: $red-medium;
+	}
+}
+}
```
Indent it please.

Suryansh5545 commented:
sir i cant make update profile scroll-able as the model is being generated at the time of click and i don't know how can i influence it .

Sanji515 commented:
> sir i cant make update profile scroll-able as the model is being generated at the time of click and i don't know how can i influence it .

You should look into the `modal component` under utility.

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html
	modified:   src/app/components/profile/profile.component.scss
	modified:   src/app/components/utility/modal/modal.component.html
	modified:   src/app/components/utility/modal/modal.component.scss
	modified:   src/styles/base.scss

Suryansh5545 commented:
i have added scroll bar and other requested thing . please check

Sanji515 commented on src/app/components/utility/modal/modal.component.html:
```diff
@@ -11,7 +11,7 @@
     <div *ngIf="isEditorRequired" class="edit-global-modal" [froalaEditor]
       [(froalaModel)]="editorContent"></div>
-    <div class="form modal-inputs">
+    <div class="form modal-inputs" >
```
Please remove this change.

Sanji515 commented on src/app/components/utility/modal/modal.component.scss:
```diff
@@ -4,3 +4,7 @@
     margin-bottom: 20px;
 }
+
+.modal-card {
```
@Suryansh5545 Can you please make this change to `base.scss` Here (https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/styles/base.scss#L1414).

Also please set it to `auto` so that it can adjust according to it's content for other models.

Suryansh5545 replied:
done sir

Suryansh5545 committed:
Delete modal.component.html

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.scss
	modified:   src/styles/base.scss

Suryansh5545 committed:
Merge branch 'master' of https://github.com/Suryansh5545/EvalAI-ngx

Suryansh5545 committed:
Delete modal.component.html

Sanji515 commented on src/app/components/utility/modal/modal.component.html:
```diff
@@ -1,42 +0,0 @@
-<div class="modal-container">
```
@Suryansh5545 Why have you deleted this file?

Suryansh5545 replied:
where isn't a need to change anything in it

Sanji515 replied:
It's scss file and you have already reverted that file.

Suryansh5545 replied:
should i repush it

Suryansh5545 replied:
i am re pushing it

Suryansh5545 replied:
i push the model.component file again

Suryansh5545 replied:
without any change from the original one

Sanji515 commented on src/styles/base.scss:
```diff
@@ -1444,6 +1444,7 @@ code {
       padding: 10px;
       text-align: right;
     }
+    overflow-y: scroll;
```
Please change it to `auto`.

Suryansh5545 replied:
sorry my mistake i forgot

Suryansh5545 committed:
modified:   src/styles/base.scss

Suryansh5545 committed:
Merge branch 'master' of https://github.com/Suryansh5545/EvalAI-ngx

Suryansh5545 committed:
new file:   src/app/components/utility/modal/modal.component.html

Sanji515 commented on src/app/components/utility/modal/modal.component.html:
```diff
@@ -1,42 +1,42 @@
-    </div>
-  </div>
-</div>
+  </div>
```
It still showing changes in this file. There shouldn't be any change in this file, Please revert this file changes.

Suryansh5545 replied:
i haven't made any changes may be github is comparing it to deleted file

Sanji515 replied:
Please revert the file again.

Suryansh5545 replied:
delte it or ?

Sanji515 replied:
Just make it as it is in the master branch of main repo(evalai-ngx).

Suryansh5545 replied:
i copied main repo file and pasted it in mine

Suryansh5545 committed:
Update modal.component.html

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -171,6 +171,30 @@ export class ProfileComponent implements OnInit {
+        },
+        {
+          isRequired: false,
+          label: 'github_url;',
```
Please remove the semi-colon `;` from label, it will give error while making back-end request

Suryansh5545 replied:
done

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -171,6 +171,30 @@ export class ProfileComponent implements OnInit {
+        },
+        {
+          isRequired: false,
+          label: 'linkedin_url;',
```
Same here

Suryansh5545 replied:
done

Suryansh5545 committed:
Update modal.component.html

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Suryansh5545 committed:
Merge branch 'master' of https://github.com/Suryansh5545/EvalAI-ngx

Sanji515 commented:
@Suryansh5545 The `Submit` button not enabling if I make any change in `linkedin url` or `github url` or `google scholar url` field. And please add some validation for a valid url in these 3 fields.

Also, when the fields are empty then on the modal input field it should show blank instead of hyphen `-`.

Sanji515 commented on src/app/components/profile/profile.component.html:
```diff
@@ -1,76 +1,126 @@
+	<div class="container-fluid main_content">
+		<div class="row">
+			<div class="dashboard">
+				<div class="md-container  main_content" id="card_container">
```
Please remove the extra space after `md-container` class name.

Suryansh5545 replied:
sir the hyphen is coming from the database because it is empty so how can i change it?

Sanji515 replied:
No, it is not coming from database but we are modifying it in our typescript here (https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/app/components/profile/profile.component.ts#L104) in order to display hyphen in our UI. So you just check if it's hyphen then send empty string `""`.

Please let me know if you didn't get this.

Suryansh5545 replied:
thx sir

Suryansh5545 replied:
i cant seem to find how can i make submit button work

Suryansh5545 replied:
and from where the comment of enter the first name is coming?

Sanji515 replied:
Please take a look how `affiliation, firstname or lastname` fields here (https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/app/components/utility/modal/modal.component.ts#L244) are being checked if they are modified or not and accordingly `Submit` button gets enabled.

Sanji515 replied:
Please let me know if you have more doubts.

Suryansh5545 replied:
sir i still cant understand this

Suryansh5545 commented:
also as said the back side is already implemented

Sanji515 commented:
@Suryansh5545 `validateModalInput` function gets called whenever any change happens in any input so first we check which input has been changed through `e.target.name` (where e is the current event which has been triggered after the change in the input field) which we have already sended here (https://github.com/Cloud-CV/EvalAI-ngx/pull/234/files#diff-fe711722081e104396ff5117802737e8R178) and then we compare `e.target.value` with the current user details like for `affiliation` >>> `this.user.affiliation`, if they are equal then it means there is no change we set `isDisabled` to `true` or `false`.

Please let me know if I'm not clear.

Suryansh5545 commented:
i figured that out but now i am stuck on url validation

Suryansh5545 commented:
in basic html if input type is url then it auto check for it but how can i do same in this

Sanji515 commented:
> in basic html if input type is url then it auto check for it but how can i do same in this

Please use regex for url validation.

Suryansh5545 commented:
i tried that if value is equal to hyphen then it become 0 but it also affects outside the update profile . how can i resolve this sir?

Sanji515 commented:
> i tried that if value is equal to hyphen then it become 0 but it also affects outside the update profile . how can i resolve this sir?

Are you trying to empty the input field instead of hyphen? Why it become 0?

Suryansh5545 commented:
sorry my mistake i am trying to put value equal to '' dual quote with blank space

Suryansh5545 commented:
> but it affects outside update profile also

Like where it affecting? It would be more better if you can push your code so that I can see what you're doing?

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html
	modified:   src/app/components/profile/profile.component.ts
	modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 commented:
i was able to complete 2 requested changes but was unable change value of hyphen

Sanji515 commented:
> i was able to complete 2 requested changes but was unable change value of hyphen

Can you please tell or push your code how you're trying to change value of hyphen?

Suryansh5545 commented:
can i remove the hyphen completely?

Suryansh5545 commented:
can i assign user[i] value to another variable and then change it ?

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts
	modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts
	modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Suryansh5545 commented:
sir i have pushed  my code with code to change hyphen to blank but it is not working

Suryansh5545 commented:
i dont know why but my uservalue is influsing the global user variable

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html
	modified:   src/app/components/profile/profile.component.ts

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Sanji515 commented:
@Suryansh5545 Instead of using another variable, how about doing this way `let first_name = (this.user['first_name'] == '-') ? '' : this.user['first_name']` for all the 6 variables?

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html
	modified:   src/app/components/profile/profile.component.ts
	modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 commented:
I have completed all requested changes

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -154,31 +152,53 @@ export class ProfileComponent implements OnInit {
+          value: this.affiliation
+        },
+        {
+          isRequired: true,
```
Can you please set back to `false`?

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -91,26 +86,30 @@ export class ProfileComponent implements OnInit {
+    let affiliation = (this.user['affiliation'] == '-') ? '' : this.user['affiliation'];
+    let google_scholar_url = (this.user['google_scholar_url'] == '-') ? '' : this.user['google_scholar_url'];
+    let github_url = (this.user['github_url'] == '-') ? '' : this.user['github_url'];
+    let linkedin_url = (this.user['linkedin_url'] == '-') ? '' : this.user['linkedin_url'];
```
I think you should move all these a little down as upto here hyphen is not assigned to the `user object`. Even I would suggest to move all these inside `updateDetails` function as these variables are not gonna used again.

Suryansh5545 replied:
little down?

Sanji515 replied:
Well, I'm suggesting to move all these inside `updateUserDetails` function.

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -7,7 +7,6 @@ import { AuthService } from '../../services/auth.service';
 import { Router, ActivatedRoute } from '@angular/router';
 import { ViewChildren, QueryList, AfterViewInit } from '@angular/core';
 import { InputComponent } from '../utility/input/input.component';
-
```
Can you please revert these deleted blank lines changes?

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Suryansh5545 commented:
i have reverted deleted line

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -120,11 +115,16 @@ export class ProfileComponent implements OnInit {
+    this.affiliation = (this.user['affiliation'] === '-') ? '' : this.user['affiliation'];
+    this.google_scholar_url = (this.user['google_scholar_url'] === '-') ? '' : this.user['google_scholar_url'];
+    this.github_url = (this.user['github_url'] === '-') ? '' : this.user['github_url'];
+    this.linkedin_url = (this.user['linkedin_url'] === '-') ? '' : this.user['linkedin_url'];
```
As these variables are not used again so better not to use them as component variable instead simply we can use them as `const` variable.

Suryansh5545 replied:
ok

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -17,48 +16,46 @@ import { InputComponent } from '../utility/input/input.component';
    * Profile completion score
    */
   pcomp: any;
-
```
I can still see these changes, please revert this and similar changes.

Suryansh5545 replied:
which changes?

Sanji515 replied:
You have these blank lines here

Suryansh5545 replied:
a blank line between each variable?

Sanji515 commented on src/app/components/utility/modal/modal.component.ts:
```diff
@@ -14,125 +13,102 @@ import { AuthService } from '../../../services/auth.service';
   styleUrls: ['./modal.component.scss']
 })
 export class ModalComponent implements OnInit {
-
```
Same here for this file also.

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Suryansh5545 commented:
i have added all the lines back

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -17,11 +16,16 @@ import { InputComponent } from '../utility/input/input.component';
+  affiliation: any;
+  google_scholar_url: any;
+  github_url: any;
+  linkedin_url: any;
```
Now we don't need variables as we defined as `const`. Please remove these

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -47,7 +51,7 @@ export class ProfileComponent implements OnInit {
    */
   isTokenModalVisible = false;
-
+  
```
Please revert this change.

Suryansh5545 replied:
but sir there are deleted lines you said me to revert

Sanji515 replied:
yeah I mean to undo that change, make it as it is in the master branch of main repo. Please remove unnecessary change which we don't need in this PR.

Suryansh5545 replied:
done sir

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -154,29 +163,52 @@ export class ProfileComponent implements OnInit {
           placeholder: 'First Name',
           type: 'text',
-          value: this.user['first_name']
+          value: this.first_name
```
Please update this and similar variable.

Sanji515 commented on src/app/components/utility/modal/modal.component.ts:
```diff
@@ -14,125 +13,102 @@ import { AuthService } from '../../../services/auth.service';
    * Input parameters object
    */
   @Input() params: any;
-
```
Please revert this and similar changes.

Suryansh5545 replied:
but sir there are deleted lines you said me to revert

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts
	modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.ts

Sanji515 commented on src/app/components/utility/modal/modal.component.ts:
```diff
@@ -255,6 +266,9 @@ export class ModalComponent implements OnInit {
         this.inputErrorMessage = 'Password do not match';
       }
     }
+    if (this.user.google_scholar_url !== this.urlval || this.user.github_url !== this.urlval || this.user.linkedin_url !== this.urlval) {
```
This doesn't seems good, please take a look how to use regex.

Suryansh5545 replied:
Where should I put it?

Suryansh5545 replied:
I can't understand

Sanji515 replied:
You can take a look to this PR (https://github.com/Cloud-CV/EvalAI-ngx/pull/231). For consistency purpose we should use the same regex.

Suryansh5545 replied:
i have implemented the regex code that was suggested in that pr but now if i write a url with correct syntax it doesn't remove please enter a valid url.

Suryansh5545 replied:
i have pushed my code

Sanji515 replied:
You also need to call `regex.test`. Like for checking `googleScholarUrl` you need to call `regex.test(googleScholarUrl)`. If it is a valid url then it will return `true`

Sanji515 replied:
Also please use this regex `^(http[s]?:\\/\\/(www\\.)?|ftp:\\/\\/(www\\.)?|www\\.){1}([0-9A-Za-z-\\.@:%_\+~#=]+)+((\\.[a-zA-Z]{2,3})+)(/(.)*)?(\\?(.)*)?`

Suryansh5545 replied:
![evalai-profileerror](https://user-images.githubusercontent.com/34577232/70844731-fa202a00-1e6a-11ea-8ba0-40f680146104.png)

Suryansh5545 replied:
it is not detecting them

Sanji515 replied:
Cause you have changed `google_scholar_url` to `googleScholarUrl` :smile:

Suryansh5545 replied:
to change this i also have to change the name variable in const params?

Suryansh5545 replied:
this code 
 const firstName = (this.user['first_name'] === '-') ? '' : this.user['first_name'];
it affecting all the variables in backend and frontend?

Suryansh5545 replied:
so i also have to change the front end html code right?

Sanji515 replied:
no it's not. It's just a temporary variable for holding the `firstName` value and then assigning it to the form variable which is `value`.

Sanji515 replied:
> so i also have to change the front end html code right?

no, you don't have to do that.

Sanji515 replied:
Also use `const` as declaration for regex.

Suryansh5545 replied:
sir it is still the same . i have pushed my code so you can look in it

Sanji515 replied:
Instead of checking on the input event change, how about checking url validation after clicking submit button in `updateUserDetails` function?

Suryansh5545 replied:
so

 if (confirm == submit ){
if(regex.test(googleScholarUrl) !== true  ){
show error message;
}
?

Suryansh5545 replied:
how will i get the value of googleScholarUrl ??

Suryansh5545 replied:
if i will pass in just googleScholarUrl then the const which we declared above will be called?

Suryansh5545 replied:
and when the submit button is pressed the form closes so there is no use of url validation in updateuserdetails

Sanji515 replied:
When the submit pressed then this (https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/app/components/utility/modal/modal.component.ts#L192) `formValidate` function gets called. Please take a look how we came to know that this call is for `Update profile` and then how can we add validation and stop from closing the modal .

Suryansh5545 replied:
sir i am this completing now before going to rest. but i am facing a problem. i have put a if condition in formvalidater . but it requires a variable to be called from profile.componet so . how can i perform that

Sanji515 replied:
Which variable is required?

Suryansh5545 replied:
like i have created a urlval var to store a true or false and i need it in model.compnent

Suryansh5545 replied:
and also what this.confirmed(this) do?

Sanji515 replied:
> and also what this.confirmed(this) do?

When validation is done then `confirmed` function gets called.

Suryansh5545 replied:
and what should i write to stop this because in the if conditions of formvalidator there is only a console.log function . so how you stop the form from being submitted?

Sanji515 replied:
You just don't have to do anything, I mean just don't call the `this.confirmed`

Suryansh5545 replied:
ok sir i got it but how i call the variable from profile.component to model.component . i tried the export method with import method but that did'nt work

Sanji515 replied:
but why you want to call some variable from `profile.component`?

Suryansh5545 replied:
Url is checked in profile.component and a true or false is assigned to a variable name urlval and it is needed for if condition in model.component . As we have to keep the url Verification in profile.component only

Sanji515 replied:
You can do the url validation in `modal.component` also.

Suryansh5545 replied:
sir i tried making the submit button stop when url is not valide but it is not working . i am pushing my code so you can take look

Suryansh5545 replied:
i found the problem sir the user.googlescholarurl is giving the test an undefined value

Suryansh5545 replied:
sir the user variable in the modal.component is containing the values that are not changing and because of this my if statement is getting a '-' hyphen in return instead of a url

Sanji515 replied:
Yeah cause it is the stored value not the one which is on the input.

Why don't you check all the url field validation here (https://github.com/Cloud-CV/EvalAI-ngx/pull/234/files#diff-102b6aae81d792250ba74b4d834949d3R195)?

Suryansh5545 replied:
how can i check for the value of url when the pre build user variable is not updating

Sanji515 replied:
Please console `this.formComponents`, you can get the current input value from there.

Suryansh5545 replied:
it is giving the input values but . it is giving them in 6 different type of input components and how can i access them. i tried this console.log(this.formComponents._results) which give 6 input component . but i am unable to access the specific value of urls fields.

Sanji515 replied:
Yes, it is an array. Just access it by index and get it's `value`. Like for `first_name` > `this.formComponents._results[0].value`

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -125,6 +125,12 @@ export class ProfileComponent implements OnInit {
+    const affiliation = (this.user['affiliation'] === '-') ? '' : this.user['affiliation'];
+    const google_scholar_url = (this.user['google_scholar_url'] === '-') ? '' : this.user['google_scholar_url'];
+    const github_url = (this.user['github_url'] === '-') ? '' : this.user['github_url'];
+    const linkedin_url = (this.user['linkedin_url'] === '-') ? '' : this.user['linkedin_url'];
```
Sorry for this :smile:  but can you please use camel-case here for variable names like `firstName`?

Suryansh5545 replied:
first_name -> FirstName
last_name -> LastName
affiliation -> Affiliation
is this okay?

Sanji515 replied:
no, they should be in camel-case, like `first_name` -> `firstName` and `last_name` -> lastName

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts
	modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.ts

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -125,6 +125,12 @@ export class ProfileComponent implements OnInit {
+    const affiliation = (this.user['affiliation'] === '-') ? '' : this.user['affiliation'];
+    const google_scholar_url = (this.user['google_scholar_url'] === '-') ? '' : this.user['google_scholar_url'];
+    const github_url = (this.user['github_url'] === '-') ? '' : this.user['github_url'];
+    const linkedin_url = (this.user['linkedin_url'] === '-') ? '' : this.user['linkedin_url'];
```
You need to camel-case for rest 3 variables also. :smiley:

Suryansh5545 replied:
googleScholarUrl or googlescholarUrl ?

Sanji515 replied:
googleScholarUrl

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts
	modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 commented:
how can change the colour of the error message displayed?

Sanji515 commented:
> how can change the colour of the error message displayed?

Don't change it for now

Suryansh5545 commented:
i made a [message] = InputErrorUrl in the model.component just affter the InputErrorMessage and declared it in the model.component.ts  also but it is not working. so should i use inpurtErrorMessage

Suryansh5545 committed:
Merge branch 'master' into master

Sanji515 commented:
> i made a [message] = InputErrorUrl in the model.component just affter the InputErrorMessage and declared it in the model.component.ts also but it is not working. so should i use inpurtErrorMessage

No, Don't add another `[message]` on the input. Please make `span` or `div` tag which will display error message just before the `Cancel` or `Submit` button. You have to toggle that div inner html and its visibility according to the validation message.

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts
	modified:   src/app/components/utility/modal/modal.component.html
	modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
Merge branch 'master' of https://github.com/Suryansh5545/EvalAI-ngx

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 commented:
sir i solved the problem and have put urlval in validatemodelinput as it will be dynamic now

Suryansh5545 commented:
and what i do about this error src/app/components/utility/modal/modal.component.ts(273,42): error TS2341: Property '_results' is private and only accessible within class 'QueryList<T>'.

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.ts

Sanji515 commented:
> and what i do about this error src/app/components/utility/modal/modal.component.ts(273,42): error TS2341: Property '_results' is private and only accessible within class 'QueryList'.

You can iterate on `this.formComponents` this way

```
this.formComponents.map((item) => {
        console.log(item.value);
      });
```

Suryansh5545 commented:
sir i can you please explain how can i use this as it is giving two outputs

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.html
	modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
Merge branch 'master' into master

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
Merge branch 'master' of https://github.com/Suryansh5545/EvalAI-ngx

Suryansh5545 committed:
modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

pushkalkatara commented:
Also, we should display an error if the URL is invalid. Currently, the pop-up shows - "something went wrong".
Ideally, the URL should be checked on the form itself.

Suryansh5545 commented:
Well i tried to do but was unable to do it with my current knowledge and the deadline i have

Suryansh5545 commented:
in the previous pr i tried to do url verification but because of the private class of _result i was unable to make it possible

Sanji515 commented on src/app/components/utility/modal/modal.component.ts:
```diff
@@ -4,6 +4,8 @@ import { GlobalService } from '../../../services/global.service';
 import { InputComponent } from '../input/input.component';
```
@Suryansh5545 Please undo all the changes of this file as we don't need them for this PR

Suryansh5545 replied:
done

Sanji515 commented on src/app/components/utility/modal/modal.component.html:
```diff
@@ -17,7 +17,7 @@
           [label]="item['label']" [placeholder]="item['placeholder']"
           [type]="item['type']" [icon]="item['icon']" [value]="item['value']"
-          [message]="inputErrorMessage" #formmodal></app-input>
+          [message]="inputErrorMessage"  #formmodal></app-input>
```
Please undo this change.

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -27,7 +27,6 @@ export class ProfileComponent implements OnInit {
    * Profile completion score
    */
   pcomp: any;
-
```
Please undo this change.

Sanji515 commented on src/app/components/profile/profile.component.ts:
```diff
@@ -16,7 +16,7 @@ import { InputComponent } from '../utility/input/input.component';
   styleUrls: ['./profile.component.scss']
 })
-export class ProfileComponent implements OnInit {
+export class ProfileComponent implements OnInit  {
```
Please undo this change also.

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts
	modified:   src/app/components/utility/modal/modal.component.html
	modified:   src/app/components/utility/modal/modal.component.ts

pushkalkatara commented on src/app/components/profile/profile.component.html:
```diff
@@ -1,76 +1,126 @@
-							</div>
-							<button class="col-lg-1 col-md-1 col-sm-1 col-xs-1 btn ev-btn-dark
-								btn-waves-effect waves-dark grad-btn code-btn"
+								</div>
```
The whitespace is not necessary i think.

Suryansh5545 replied:
i cant understand?

pushkalkatara commented on src/app/components/profile/profile.component.html:
```diff
@@ -1,76 +1,126 @@
-							<button class="col-lg-1 col-md-1 col-sm-1 col-xs-1 btn ev-btn-dark
-								btn-waves-effect waves-dark grad-btn code-btn"
+								</div>
+								<button class="col-lg-1 col-md-1 col-sm-1 col-xs-1 btn ev-btn-dark
```
Same here, the whitespace is not required.

Sanji515 reviewed (commented):
On smaller screen `Get Auth Token` model button alignment doesn't look good and also horizontal scroll shows up which should not be the expected behaviour. @Suryansh5545 Please fix this.

pushkalkatara commented on src/app/components/profile/profile.component.scss:
```diff
@@ -1,82 +1,83 @@
 
-.modal-container{
-	.modal-card {
+.modal-container {
```
Please fix the whitespace.

pushkalkatara commented on src/app/components/profile/profile.component.scss:
```diff
@@ -1,82 +1,83 @@
-.modal-container{
-	.modal-card {
+.modal-container {
+	.modal-card { 
```
Please fix this whitespace.

Sanji515 commented on src/app/components/profile/profile.component.html:
```diff
@@ -1,76 +1,126 @@
-								btn-waves-effect waves-dark grad-btn code-btn"
+								</div>
+								<button class="col-lg-1 col-md-1 col-sm-1 col-xs-1 btn ev-btn-dark
+								btn-waves-effect waves-dark grad-btn code-btn" id="auth_butt_style"
```
@Suryansh5545 Can we please change this `id` name which can make some sense?

pushkalkatara commented on src/app/components/profile/profile.component.scss:
```diff
@@ -1,82 +1,83 @@
 			margin-top: 15px;
 		}
 	}
+	
```
Last one i think. Please remove this whitespace.

Sanji515 commented on src/app/components/utility/modal/modal.component.html:
```diff
@@ -39,4 +39,4 @@
     </div>
   </div>
-</div>
+</div>
```
@Suryansh5545 Please add a new blank here.

Sanji515 commented on src/app/components/utility/modal/modal.component.ts:
```diff
@@ -260,4 +260,4 @@ export class ModalComponent implements OnInit {
     this.isDisabled = e.target.value === '';
   }
-}
+}
```
Same here also (a blank line)

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html
	modified:   src/app/components/profile/profile.component.scss
	modified:   src/app/components/utility/modal/modal.component.html
	modified:   src/app/components/utility/modal/modal.component.ts

Suryansh5545 commented:
@Sanji515 sir the horizontal scroll bar look fine in my screen and get auth token button look fit in small screen as in evalai . can you send me a pic

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.ts

Sanji515 commented:
> @Sanji515 sir the horizontal scroll bar look fine in my screen and get auth token button look fit in small screen as in evalai . can you send me a pic

![Screenshot from 2019-12-16 16-37-15](https://user-images.githubusercontent.com/32524438/70903892-ec62d400-2025-11ea-8ebb-d7292c4c3af1.png)

@Suryansh5545 Please take a look, I've scrolled a bit horizontally then taken screenshot.

Suryansh5545 commented:
the form is cutting down. so should i make it scroll able?

Suryansh5545 commented:
i have to change the button align ?

Sanji515 commented:
> i have to change the button align ?

On smaller screens as shown on the above screenshot it doesn't look good, please make it better.

Suryansh5545 commented:
![Selection_026](https://user-images.githubusercontent.com/34577232/70905912-1f5b9680-202b-11ea-88a4-7487b9d669d3.png)

Suryansh5545 commented:
i have made the icon come close to the text bar

Suryansh5545 commented:
i tried to make the button move up but it is not possible in so compact space

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.scss

Sanji515 commented:
![Screenshot from 2019-12-16 18-05-11](https://user-images.githubusercontent.com/32524438/70907345-a8280180-202e-11ea-8a50-1c1df942a32e.png)


@Suryansh5545 Can you please fix this(^) too on smaller screens?

Suryansh5545 commented:
Sir the button will get out of screen if width is changed

Sanji515 commented:
> Sir the button will get out of screen if width is changed

No, not width but the button height should automatically adjust according to the text size on smaller screens.

Suryansh5545 commented:
![Selection_030](https://user-images.githubusercontent.com/34577232/70908449-2eddde00-2031-11ea-9a40-f37cf99140fe.png)
At hight 240px.

![Selection_029](https://user-images.githubusercontent.com/34577232/70908471-3bfacd00-2031-11ea-8fe7-5cef9cbbcb8d.png)
At hight 740px.
![Selection_028](https://user-images.githubusercontent.com/34577232/70908500-4cab4300-2031-11ea-8bf1-de566fbc1bf5.png)
at height 1200px.



The Button is allowed to resize it self after 1270px so instead of showing only get your auth in sceen size of 1200px is will show full message no matter how small screen is.

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.html
	modified:   src/app/components/profile/profile.component.scss

Sanji515 commented:
Hey @Suryansh5545 This doesn't look good. Can you try decreasing the `line-height` of text and add some `top` and `bottom` padding to the button for `max-width >= 600 to 1270`? Also do the same for extra small devices.

Suryansh5545 commented:
i have completed the changes

Suryansh5545 committed:
modified:   src/app/components/profile/profile.component.scss

Sanji515 reviewed (approved):
Thanks @Suryansh5545 for all this hard work and for responding quickly to the comments :+1:  LGTM, this can be merged now :tada: 

Note: This PR is about modifying the profile page to support the url fields (which are static in this PR).

Suryansh5545 commented:
Thank you Sir for helping me

Suryansh5545 commented:
If any change is required I will be happy to deliver it

Suryansh5545 committed:
Revert "	modified:   src/app/components/challenge/challenge.component.html"

This reverts commit 2e15a2d46fbe0f40420657cf5f1caf5dc089a07c.

	modified:   src/app/components/challenge/challenge.component.html
	modified:   src/app/components/challenge/challenge.component.scss
	modified:   src/app/components/challenge/challengesubmit/challengesubmit.component.html
	modified:   src/app/components/challenge/challengesubmit/challengesubmit.component.scss
	modified:   src/app/components/challenge/challengesubmit/challengesubmit.component.ts

Suryansh5545 committed:
Revert "	modified:   src/app/components/challenge/challenge.component.html"

This reverts commit 2e15a2d46fbe0f40420657cf5f1caf5dc089a07c.

	modified:   src/app/components/challenge/challenge.component.html
	modified:   src/app/components/challenge/challenge.component.scss
	modified:   src/app/components/challenge/challengesubmit/challengesubmit.component.html
	modified:   src/app/components/challenge/challengesubmit/challengesubmit.component.scss
	modified:   src/app/components/challenge/challengesubmit/challengesubmit.component.ts

drepram commented:
Hi @Suryansh5545, I forked your improved version of EvalAI-ngx and I've cleaned up some of the code and fixed the logic in it. Please take a look at #243.
